yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +118,84 @@ impl Clone for VTag {
     }
 }
 
+/// HTML output for a VTag is not necessarily deterministic due to the
+/// serialization of props which do not have a particular ordering.
+#[cfg(feature = "ssr")]
+impl TryFrom<VTag> for Html {
+    type Error = HtmlStringifyError;
+
+    fn try_from(value: VTag) -> Result<Html, HtmlStringifyError> {
+        let mut parts: Vec<String> = Vec::new();
+        let tag_name = htmlescape::encode_minimal(&value.tag).to_lowercase();
+
+        for c in tag_name.chars() {
+            let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+            if !is_alnum && c != '-' && c != '_' && c != ':' {
+                return Err(HtmlStringifyError::InvalidTagName(tag_name));
+            }
+        }
+
+        parts.push(format!("<{}", tag_name).to_string());
+
+        for (key_unclean, value) in &value.attributes {
+            let key = key_unclean.to_lowercase();
+            // checked, value (special if textarea), disabled, href?, selected,
+            // kind -> type if input, disallow ref, disallow LISTENER_SET, class
+
+            for c in key.chars() {
+                let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+                if !is_alnum && c != '-' && c != '_' && c != ':' {
```

Comment:
This might just be me, but I suspect that if a string is alphanumeric then the condition will never be true `c != '-' && c != '_' && c != ':'`
```suggestion
                if !c.is_alphanumeric() {
```

Reply:
Oops, my bad – that's not true.